lib/streamlit/elements/lib/options_selector_utils.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import (
+    Any,
+    Sequence,
+    cast,
+)
+
+from streamlit.dataframe_util import OptionSequence, convert_anything_to_sequence
+from streamlit.errors import StreamlitAPIException
+from streamlit.type_util import (
+    T,
+    check_python_comparable,
+    is_type,
+)
+
+
+def check_and_convert_to_indices(
```

Comment:
This method will need some refactoring to actually use `convert_anything_to_sequence` instead of the custom logic here. But I can take that on as a follow up PR.

Reply:
This is the logic from `multiselect` that has existed for a couple of years now, its just moved to another function to share with `button group`. Refactoring it in a follow-up PR sounds good to me 👍